Color Table Window Overhaul May 2023 (#18691)
visit-dav/visit: src/doc/using_visit/MakingItPretty/Color_tables.rst

   The color table window

You can open VisIt_'s **Color table window**, shown in :numref:`Figure %s <fig-MakingItPretty-ColorTables>`, by selecting **Color table** from the **Main Window**'s **Controls** menu, or by pressing ``ctrl`` + ``t``.
The **Color table window** is separated into three areas: The top area allows you to set the default color tables. 
The right-hand side, or the manager portion of the window, allows you to create, delete, and export new color tables, as well as organize all the available color tables.
See :ref:`Filtering_With_Tags` for more information. 

> The left-hand side, or editor portion of the window, allows you to edit color tables by adding, removing, moving, or changing the color of color control points.

Reviewer: ```suggestion
The left-hand side, or editor portion of the window, allows you to edit color tables by adding, removing, moving, or changing the color of control points.
```

> A color control point is a point with a color that influences how the color table will look.

Reviewer: ```suggestion
A color control point is a point with a color that influences how color varies through the table.
```

Searching for Color Tables
~~~~~~~~~~~~~~~~~~~~~~~~~~

> Another option for finding the right color table is to use the search bar.

Reviewer: ```suggestion
Another option for finding the right color table is to use the search feature.
```

> With this option, users can type a search term into the **Search** text box, and only color tables that have a name containing that term will appear in the color table name table.

Reviewer: Does the search feature match only text in name? Or, does it also match text in tags? Is the matching case-sensitive?

My prefs...case insensitve matching of either the name of the color table or the tag(s) associated with it.

Do we support any remarks associated with a color table? Like, `"This color table is designed for faciliating data differencing."` If so, maybe matching that text via search as well.

Author: > Does the search feature match only text in name?

The search feature only matches text in the color table names.

> Or, does it also match text in tags?

We do not support searching for color tables that have tags that match the search.

> Is the matching case-sensitive?

The matching is case-insensitive.

Reviewer: Works for me.

Author: > My prefs...case insensitve matching of either the name of the color table or the tag(s) associated with it.

I think matching on the tags could get confusing. The search feature is designed to be used in tandem with the tag filtering. So, you can choose your favorite tags and then use search to whittle down the results more, or the opposite: you could start with a search and then use tags to get that list down to the results you want. I think having them behave independently of one another makes the system a bit easier to use.

Author: > Do we support any remarks associated with a color table?

Right now we don't support remarks for color tables. I think we talked about adding that in the past. I'll add that idea to the running list I have #18136

> Prior to VisIt_ 3.4, the **Name** text box doubled as the search bar, and users had to check the box for it to the right of the **Name** text box.

Reviewer: Don't think this is sentance is needed.

Author: Ok. I was worried about having documentation that does not reflect the VisIt that users are using, but I suppose they can look at earlier versions of the docs if they want to know what is true for their version of visit.

> As of VisIt_ 3.4, searching is always on.

Reviewer: Hmmm...what does *always on* mean? I think it means it behaves like similar search text works elswhere...empty/blank text matches anything whereas non-empty matches whatever it matches. And, I don't think that actually requires any elaboration here. So, I would remove this line too.

Users need only edit the search bar contents to use it.

> To disable searching, simply delete all text in the search bar.

Reviewer: 528 and 529 here are fine to include. I can't recall, does an empty search box have greyed out text that hints at what to do? like Enter search text here?

Author: It does not have any text in it. I could look into adding that, but I am inclined to leave as is. I feel that, with the "search" label next to it, it is already clear what it does: https://github.com/visit-dav/visit/pull/18691/files#diff-073a58fb3f01e4369e71ff89c1c8fc93923bab1db18c2c1ad47a3ff47b0d6077

Reviewer: Thats fine too. I see that kind of thing, for example, all over GitHub's browser GUI.

> Prior to VisIt_ 3.4, while searching was enabled, searching happened automatically each time the **Name** text box was edited.

Reviewer: Agian, not necessary to have this. I would remove it through 534.
